controllers/provider-aws/pkg/controlplane/internal/helper.go

```diff
@@ -0,0 +1,235 @@
+					},
+					KubeConfigRequest: &secrets.KubeConfigRequest{
+						ClusterName:  clusterName,
+						APIServerURL: common.KubeAPIServerDeploymentName,
+					},
+				},
+				&secrets.ControlPlaneSecretConfig{
+					CertificateSecretConfig: &secrets.CertificateSecretConfig{
+						Name:       cloudControllerManagerServerName,
+						CommonName: cloudControllerManagerDeploymentName,
+						DNSNames:   controlplane.DNSNamesForService(cloudControllerManagerDeploymentName, clusterName),
+						CertType:   secrets.ServerCert,
+						SigningCA:  cas[gardencorev1alpha1.SecretNameCACluster],
+					},
+				},
+			}
+		},
+	}
+
+	configChart = &chart.Chart{
+		Name: "cloud-provider-config",
+		Path: filepath.Join(aws.InternalChartsPath, "cloud-provider-config"),
+		Objects: []*chart.Object{
+			{Type: &corev1.ConfigMap{}, Name: common.CloudProviderConfigName},
+		},
+	}
+
+	controlPlaneChart = &chart.Chart{
+		Name:   "cloud-controller-manager",
+		Path:   filepath.Join(aws.InternalChartsPath, "cloud-controller-manager"),
+		Images: []string{common.HyperkubeImageName},
+		Objects: []*chart.Object{
+			{Type: &corev1.Service{}, Name: "cloud-controller-manager"},
+			{Type: &appsv1.Deployment{}, Name: "cloud-controller-manager"},
+		},
+	}
+)
+
+// getConfigChartValues collects and returns the configuration chart values.
+func getConfigChartValues(decoder runtime.Decoder, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) (map[string]interface{}, error) {
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
+	if _, _, err := decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
+		return nil, errors.Wrapf(err, "could not decode providerConfig of controlplane '%s'", util.ObjectName(cp))
+	}
+
+	// Decode infrastructureProviderStatus
+	infraStatus := &apisaws.InfrastructureStatus{}
+	if _, _, err := decoder.Decode(cp.Spec.InfrastructureProviderStatus.Raw, nil, infraStatus); err != nil {
+		return nil, errors.Wrapf(err, "could not decode infrastructureProviderStatus of controlplane '%s'", util.ObjectName(cp))
+	}
+
+	// Determine subnet ID and zone
+	subnetID, zone, err := getSubnetIDAndZone(infraStatus)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not determine subnet ID or zone from infrastructureProviderStatus of controlplane '%s'", util.ObjectName(cp))
+	}
+
+	// Collect config chart values
+	return map[string]interface{}{
+		"cloudProviderConfig": getCloudProviderConfig(infraStatus.VPC.ID, subnetID, zone, cp.Namespace),
+	}, nil
+}
+
+// getControlPlaneChartValues collects and returns the control plane chart values.
+func getControlPlaneChartValues(decoder runtime.Decoder, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster, checksums map[string]string) (map[string]interface{}, error) {
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
+	if _, _, err := decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
```

Comment:
Write a `ControlPlaneConfigFromProviderConfig` method using a 'constant' scheme in a separate file just as we did it for the gcp infrastructure controller. Then this function only has a single responsibility.